contracts/OperatorStage2.sol

```diff
@@ -48,12 +65,33 @@ contract OperatorStage2 is AccessControl {
 		tokenGateway = StandardTokenGateway(gatewayAddress);
 	}
 
+	// Adds a new rate that allows a user to bond with
+	function addRate(uint256 _rate, uint256 _maturityInWeeks) public onlyOperatorStage2 {
+		allowedYieldToWeeks[_rate] = _maturityInWeeks;
+		BondRate memory br = BondRate(_rate, _maturityInWeeks);
+		ratesAvailable.push(br);
+		emit Yield(_rate, _maturityInWeeks);
+	}
+
+	// Sets a rate to zero, not removing it but making it obsolete
+	// TODO: make this nicer and not just bond for 0% yield
+	function removeRate(uint256 _rate) public onlyOperatorStage2 {
+		allowedYieldToWeeks[_rate] = 0;
+		emit Yield(_rate, 0);
+	}
+
+	// Displays all the rates available as pairs of (rate, duration)
+	function showRates() public view returns(BondRate[] memory) {
+		return ratesAvailable;
+	}
+
 	function newBond(
 		address _user,
 		uint256 _amountSeuro,
 		uint256 _weeks,
 		uint256 _rate
-	) public onlyOperatorStage2 {
+	) public {
+		require(allowedYieldToWeeks[_rate] > 0, "err-missing-rate");
```

Comment:
any reason that we're still sending in both weeks + rate? rather than just sending in weeks then getting the rate for it inside the contract? not against this way, just interested why we'd add more logic to the frontend

Reply:
good spot, since we will retrieve it anyway